File: node-http/models/groupInfo.js
/**
 * @Author:acexy@thankjava.com
 * 18/8/20
 * @Description:
 */
const baseModel = require('./basic.js');

class groupInfo extends baseModel {

    Schema() {
        return {
            groupNickname: {type: String, required: true},
            groupName: {type: String, required: true, unique: true},
            ownerUserId: {type: String, required: true},
            introduction: {type: String, required: true},
            headImg: {type: String, required: false},
        };
    };

    SchemaName() {
        return 'group_info';
    };

    save(groupInfo) {
        return new this.model(groupInfo).save();
    }

    selectByGroupName(groupName) {
        return this.model.findOne({groupName: groupName}).exec();
    }

    selectByConditionOnlyOne(friendRelation) {
        return this.model.findOne(friendRelation).exec();
    }

    updateById(id, friendRelation) {
        return this.model.updateOne({_id: baseModel.typeObject(id)}, friendRelation).exec();
    }

    selectByIds(ids) {
        let queryIds = [];
        for (let index in ids) {
            queryIds.push(baseModel.typeObject(ids[index]));
        }
        return this.model.find({_id: {"$in": queryIds}}).exec();
    }

    selectByOwnerUserId(ownerUserId) {
        return this.model.find({ownerUserId: ownerUserId}).exec()
    }

    selectByGroupNicknameOrGroupName(groupName, groupNickname) {
        let query = {};
        if (groupName && groupNickname) {
            query = {"$or": [{groupName: groupName}, {groupNickname: {"$regex": groupNickname, "$options": 'i'}}]};
        } else {
            if (groupName) query.groupName = groupName;
            if (groupNickname) query.groupNickname = {"$regex": groupNickname, "$options": 'i'};
        }
        return this.model.find(query).exec();
    }
}

module.exports = new groupInfo();

File: java-websocket/src/main/java/com/thankjava/wchat/db/entity/GroupInfo.java
package com.thankjava.wchat.db.entity;

/**
 * @Author: acexy@thankjava.com
 * 2018/8/14
 * @Description:
 **/
public class GroupInfo extends BaseEntity {

    // 群组名
    private String groupNickname;

    // 群帐号
    private String groupName;

    // 群主
    private String ownerUserId;

    // 简介
    private String introduction;

    // 头像
    private String headImg;


    public String getGroupNickname() {
        return groupNickname;
    }

    public void setGroupNickname(String groupNickname) {
        this.groupNickname = groupNickname;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public String getIntroduction() {
        return introduction;
    }

    public void setIntroduction(String introduction) {
        this.introduction = introduction;
    }

    public String getHeadImg() {
        return headImg;
    }

    public void setHeadImg(String headImg) {
        this.headImg = headImg;
    }

    public String getOwnerUserId() {
        return ownerUserId;
    }

    public void setOwnerUserId(String ownerUserId) {
        this.ownerUserId = ownerUserId;
    }
}
